lib/translation-blender.js

```diff
@@ -0,0 +1,139 @@
+/* jshint node: true */
+
+/**
+* Copyright 2015, Yahoo! Inc.
+* Copyrights licensed under the New BSD License. See the accompanying LICENSE file for terms.
+*/
+
+'use strict';
+
+var CachingWriter = require('broccoli-caching-writer');
+var walkSync      = require('walk-sync');
+var yaml          = require('js-yaml');
+var mkdirp        = require('mkdirp');
+var extend        = require('extend');
+var chalk         = require('chalk');
+var glob          = require('glob');
+var path          = require('path');
+var fs            = require('fs');
+
+var log = console.log;
+
+/**
+* Turns an object into a single dimensional array of strings
+*
+* propKeys({ a: true, b: { c: true }}) => ["a", "b.c"]
+*
+* NOTE" Period within a key are escaped.
+* ie, `propKeys({ 'a.a': true, b: { c: true }})` => `["a\.a", "b.c"]`
+*
+* @method propKeys
+* @param {Object} object
+* @return {Array} Returns array of strings
+* @private
+*/
+function propKeys (object) {
+    var result = [];
+    var escaped;
+
+    for (var key in object) {
+        escaped = key.replace(/\./g, '\\.');
+
+        if (object.hasOwnProperty(key)) {
+            if (typeof object[key] === 'object') {
+                result = result.concat(propKeys(object[key]).map(function (_key) {
+                    return escaped + '.' + _key;
+                }));
+            } else {
+                result.push(escaped);
+            }
+        }
+    }
+
+    return result;
+}
+
+function TranslationBlender (inputTrees, options) {
```

Comment:
To anyone reviewing, the goal here is to take the default translation file and compares other translation files against the default.  It then injects and missing keys in from the default translation.

```
// en.json
{
"foo": "bar",
"bar": "baz"
```

```
// fr.json
{
"foo": "bar"
```

becomes 

```
// fr.json
{
"foo": "bar",
"bar": "baz"
```